llvm/lib/Target/Sparc/SparcISelLowering.cpp

```diff
@@ -3599,6 +3615,15 @@ bool SparcTargetLowering::isFPImmLegal(const APFloat &Imm, EVT VT,
 
 bool SparcTargetLowering::isCtlzFast() const { return Subtarget->isVIS3(); }
 
+bool SparcTargetLowering::isCheapToSpeculateCttz(Type *Ty) const {
+  // We lack native cttz, however,
+  // On 64-bit targets it is cheap to implement it in terms of popc.
+  if (Subtarget->is64Bit() && Subtarget->usePopc())
+    return true;
+  // Otherwise, implementing cttz in terms of ctlz is still cheap.
+  return isCheapToSpeculateCtlz(Ty);
```

Comment:
Not sure where did my comment go, I'll repeat just in case.
Isn't expansion in terms of ctlz cheaper than that in terms of ctpop? (read: Shouldn't the comments be swapped?)

Reply:
Never mind, I saw it in tests :)